minorminer/utils/raster_embedding.py

```diff
@@ -11,13 +11,81 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import warnings
+
+import matplotlib.pyplot as plt
 
-import networkx as nx
 import dwave_networkx as dnx
+import networkx as nx
 import numpy as np
-import warnings
+
 from minorminer.subgraph import find_subgraph
 
+def visualize_embeddings(H, embeddings=None, title=None, **kwargs):
+    """Visualizes the embeddings using dwave_networkx's layout functions.
+
+    Args:
+        H (networkx.Graph): The input graph to be visualized. If the graph 
+            represents a specialized topology, it must be constructed using 
+            dwave_networkx (e.g., chimera, pegasus, or zephyr graphs).
+        embeddings (list of dict, optional): A list of embeddings where each 
+            embedding is a dictionary mapping nodes of the source graph to 
+            nodes in the target graph. If not provided, only the graph `H` 
+            will be visualized without specific embeddings.
+        title (str, optional): Title of the plot. Defaults to None.
+            **kwargs: Additional keyword arguments passed to the drawing functions 
+            (e.g., node size, font size).
+    Draws:
+        - Specialized layouts: Uses dwave_networkx's `draw_chimera`, 
+          `draw_pegasus`, or `draw_zephyr` if the graph family is identified.
+        - General layouts: Falls back to networkx's `draw_networkx` for 
+          graphs with unknown topology.
+    """
+    fig, ax = plt.subplots(figsize=(10, 8))
+
+    # Create node color mapping
+    node_color_dict = {q: 'grey' for q in H.nodes()}
+    if embeddings:
+        cmap = plt.get_cmap('viridis')
+        norm = plt.Normalize(vmin=0, vmax=len(embeddings) - 1)
+        node_color_dict.update({q: cmap(norm(idx)) for idx, emb in enumerate(embeddings) for q in emb.values()})
+
+    # Create edge color mapping
+    edge_color_dict = {}
+    for v1, v2 in H.edges():
+        if node_color_dict[v1] == node_color_dict[v2]:
+            edge_color_dict[(v1, v2)] = node_color_dict[v1]
+        else:
+            edge_color_dict[(v1, v2)] = 'grey'
+
+    # Default drawing arguments
+    draw_kwargs = {
+        'G': H,
+        'node_color': [node_color_dict[q] for q in H.nodes()],
+        'edge_color': [edge_color_dict[e] for e in H.edges()],
+        'node_shape': 'o',
+        'ax': ax,
+        'with_labels': False,
```

Comment:
use `"` instead of `'` (PEP 8). this comment is applicable to other strings

Reply:
When I check other ocean code base, these arguments seems to be declared with single quote, which is similar to what we did in the shimming tutorial. Doesn't PEP 8 allows both ``` " ``` and ``` ' ``` for strings as long as the format is consistent?

Reply:
Oooh, thanks for catching that. Let's stick to single-quotes